docs/DeveloperGuide.md

```diff
@@ -1,38 +1,662 @@
 # Developer Guide
 
+## Table of Contents
+
+* [Acknowledgements](#acknowledgements)
+* [Introduction](#introduction)
+* [Design](#design)
+* [Commands and Implementation](#commands-and-implementation)
+* [Appendix: Requirements](#appendix-requirements)
+* [Appendix: Manual Testing](#appendix-manual-testing)
+* [Frequently Asked Questions](#frequently-asked-questions)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Our team has referenced [Address Book (Level-3)](https://github.com/se-edu/addressbook-level3) referenced for their [User Guide (UG)](https://se-education.org/addressbook-level3/UserGuide.html) and [Developer Guide (DG)](https://se-education.org/addressbook-level3/DeveloperGuide.html) to better structure our own Developer Guide.
+
+- The `java.util.Scanner` class from the Java Standard Library is used for reading user input.
+- The `JUnit 5` testing framework is used for writing and running unit tests.
+- {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+
+## Introduction
+
+The purpose of this guide is to provide an explanation for all the functions and internal workings in PulsePilot. This enables any technical readers to get a detailed understanding of the application's implementation, making it easier for them to contribute to the project or adapt it according to their preferences.
+
+###### [Back to table of contents](#table-of-contents)
+
+## Design
+
+* [Overview of Components](#overview-of-components)
+* [UI](#ui-package)
+    * [Handler](#handler)
+    * [Output](#output)
+* [Workout](#workout-package)
+    * [WorkoutList](#workout-list)
+    * [Gym](#gym)
+        * [GymStation](#gym-station)
+        * [GymSet](#gym-set)
+    * [Run](#run)
+* [Health](#health-package)
+    * [HealthList](#health-list)
+    * [Bmi](#bmi)
+    * [Period](#period)
+    * [Appointment](#appointment)
+* [Utility](#utility-package)
+    * [Parser](#parser)
+    * [Validation](#validation)
+    * [CustomExceptions](#custom-exceptions)
+    * [Filters](#filters)
+* [Storage](#storage-package)
+    * [LogFile](#log-file)
+    * [DataFile](#data-file)
+* [PulsePilot](#pulsepilot-package)
+* [Constants](#constants-package)
+
+### Overview of Components
+
+The application follows an Object-Oriented Design approach, with separate classes for handling different components of the application, such as user input, output, exercise logging, and health data management.
+
+
+![ArchitectureDiagram](![img.png](img/architecture_diagram.png)
```

Comment:
<img width="899" alt="Screenshot 2024-04-04 at 1 20 52 PM" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/110795904/0aa9309d-4727-4d42-bba0-a923d3e61aa2">

Alignment issue on the Architecture diagram.